src/ServiceControl.Audit.Persistence.RavenDb5/RavenDbPersistenceConfiguration.cs

```diff
@@ -1,74 +1,35 @@
 ﻿namespace ServiceControl.Audit.Persistence.RavenDb
 {
     using System;
-    using System.Threading;
-    using System.Threading.Tasks;
-    using Microsoft.Extensions.DependencyInjection;
     using NServiceBus.Logging;
-    using Persistence.UnitOfWork;
     using ServiceControl.Audit.Persistence.RavenDb5;
-    using UnitOfWork;
 
     public class RavenDbPersistenceConfiguration : IPersistenceConfiguration
     {
-        public IPersistenceLifecycle ConfigureServices(IServiceCollection serviceCollection, PersistenceSettings settings)
+        public IPersistence Create(PersistenceSettings settings)
         {
-            var persistenceLifeCycle = CreatePersistenceLifecycle(settings);
-
-            serviceCollection.AddSingleton(settings);
-            serviceCollection.AddSingleton<IRavenDbSessionProvider, RavenDbSessionProvider>();
-            serviceCollection.AddSingleton<IAuditDataStore, RavenDbAuditDataStore>();
-            serviceCollection.AddSingleton<IAuditIngestionUnitOfWorkFactory, RavenDbAuditIngestionUnitOfWorkFactory>();
-            serviceCollection.AddSingleton<IFailedAuditStorage, RavenDbFailedAuditStorage>();
-            serviceCollection.AddSingleton<IRavenDbDocumentStoreProvider>(_ => persistenceLifeCycle);
+            var dataBaseConfiguration = GetDatabaseConfiguration(settings);
+            var expirationProcessTimerInSeconds = GetExpirationProcessTimerInSeconds(settings);
+            var databaseSetup = new DatabaseSetup(expirationProcessTimerInSeconds, settings.EnableFullTextSearchOnBodies, dataBaseConfiguration);
+            var lifecycle = CreateLifecycle(settings, dataBaseConfiguration);
```

Comment:
I understand what's going on. However, something seems to be off with the naming. We're creating the lifecycle here and not in `CreateLifecyle` where instead we're only configuring DI.

Shouldn't the lifecycle creation be moved to the `RavenDb5Persistence` class?

Reply:
Agreed, I have now moved it